docs/sources/index.md

```diff
@@ -293,5 +292,44 @@ TypesafeConfig.fromHoconString(jsonString, automaticDescription)
 
 
 ```
-
 Please check other ways to load the hocon file in `TypesafeConfig`
+
+## Command Line Arguments Source
+Simply source config from command line arguments
+```scala mdoc:silent
+final case class SimpleCommandLineConfig(key1: String, key2: String)
+
+val simpleCmdLineArgs = "--key1 value1 --key2 value2"
+val simpleSource = ConfigSource.fromCommandLineArgs(simpleCmdLineArgs.split(' ').toList)
+val simpleConfig = descriptor[SimpleCommandLineConfig] from simpleSource
+```
+With nested path in command line arguments
```

Comment:
Remove line number 306, and add the below heading - its a subheading, meaning , `###`

### Nested config using command line arguments: Approach 1

```
 final case class SparkConf(key1: String, key2: String)
 final case class NestedCommandLineConfig(conf: SparkConf, key3: String)

 val nestedCmdLineArgs = "--conf.key1 v1 --conf.key2 v2 --key3 v3"
 
val nestedSource = 
   ConfigSource.fromCommandLineArgs(
     nestedCmdLineArgs.split(' ').toList, 
     keyDelimiter = Some('.')
  )

 val nestedConfig = descriptor[NestedCommandLineConfig] from nestedSource
 assert (read(nestedConfig)  == Right(<your result here>)

```
This config is for those dev who really used to system properties (-Dconf.key=1) and want to take the same approach towards command line arguments. Here we make use of delimiter `.` as the tool to nesting.

For those who hate delimited keys in command-line and the associated nesting, we will have different approach as given below

### Nested config using command line arguments: Approach 2

```
 val nestedCmdLineArgs2 = "--conf -key1=v1 --conf -key2=v2 --key3 v3"

 val nestedSource2 = 
  ConfigSource.fromCommandLineArgs(nestedCmdLineArgs.split(' ').toList)

 val nestedConfig2 = descriptor[NestedCommandLineConfig] from nestedSource2

 assert (read(nestedConfig2)  == Right(<your result here>)

```
Here we don't use delimiters for nesting, hence `keyDelimiter` is None. In this case any key value that comes after `--conf` comes under the root path `conf`. This is followed in various places such as spark.

In fact, we can go any level nesting. For example,  we can give  `---aws --db -url="v" ---aws --kinesis -topic=x`, although let's don't complicate our command line arguments.

### Map using command line arguments
Both the approaches that we saw with `nesting` is applicable to map.

```scala
val args = --conf.key1=value1  --conf.key2=value2

`map("conf")(string)` retrieving `Map("key1" -> "value1", "key2" -> "value2")`

```
This will also work if args is `--conf -key1=value1 --conf -key2=value2`